src/blocks/privacy/pipewire.rs

```diff
@@ -151,20 +156,29 @@ impl Client {
         main_loop.run();
     }
 
-    fn add_event_listener(&self, tx: UnboundedSender<()>) {
-        self.event_listeners.lock().unwrap().push(tx);
+    fn add_event_listener(&self, notify: &Arc<Notify>) {
+        self.event_listeners
+            .lock()
+            .unwrap()
+            .push(Arc::downgrade(notify));
     }
 
     fn send_update_event(&self) {
         self.event_listeners
             .lock()
             .unwrap()
-            .retain(|tx| tx.send(()).is_ok());
+            .retain(|notify| notify.upgrade().inspect(|x| x.notify_one()).is_some());
     }
 
     async fn wait_until_ready(&self) {
-        while !*self.ready.lock().unwrap() {
-            sleep(Duration::from_millis(1)).await;
+        // Call `.notified()` _before_ checking `ready` to avoid a race condition:
+        // 1. We check `ready` and it is `false`.
+        // 2. `ready` is set to `true` and subscribers are notified by another thread.
+        // 3. We subscribe but are never notified.
+        let notify = self.ready_notify.notified();
+        atomic::fence(atomic::Ordering::SeqCst);
```

Comment:
Can you explain why the fence is required?

My understanding is that the fence would allow for the load to have `Relaxed` ordering.

Also why `SeqCst` instead of `AcqRel`?

Reply:
> Can you explain why the fence is required?

To make sure the atomic load in `.notified()` happens before our `ready.load()`. That is if understand them correctly :) Oh, and I forgot a fence when notifying that roundtrip is done.

> My understanding is that the fence would allow for the load to have Relaxed ordering.
>
> Also why SeqCst instead Release on the store and Acquire on the load?

Admittedly I'm not very familiar with memory orderings, so I've just used the strongest one. And `.notified()` also uses `SeqCst`.

Reply:
I removed this